apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/main/java/gov/cms/bfd/pipeline/ccw/rif/extract/s3/S3Utilities.java

```diff
@@ -24,7 +35,36 @@ public static AmazonS3 createS3Client(ExtractionOptions options) {
    * @return the {@link AmazonS3} client to use
    */
   public static AmazonS3 createS3Client(Regions awsS3Region) {
+    Boolean s3LocalDev = Boolean.parseBoolean(System.getProperty("s3.local", "false"));
+
+    if (s3LocalDev) {
+      return createS3MinioClient(awsS3Region);
+    }
+
     AmazonS3 s3Client = AmazonS3ClientBuilder.standard().withRegion(awsS3Region).build();
     return s3Client;
   }
+
+  /**
+   * @param awsS3Region the AWS {@link Regions} that should be used when interacting with S3
+   * @return the {@link AmazonS3} minio client to use
+   */
+  public static AmazonS3 createS3MinioClient(Regions awsS3Region) {
+    S3MinioConfig minioConfig = S3MinioConfig.Singleton();
+
+    AWSCredentials credentials =
+        new BasicAWSCredentials(minioConfig.minioUserName, minioConfig.minioPassword);
+
+    ClientConfiguration clientConfiguration = new ClientConfiguration();
+    clientConfiguration.setSignerOverride("AWSS3V4SignerType");
```

Comment:
Maybe a comment here explaining what this is and why it is needed?

Reply:
done